src/main/java/com/meraphorce/services/UserService.java

```diff
@@ -80,22 +71,17 @@ public List<UserDTO> getAllUsers() {
      * @author Juan Chavez
      * @since May/14/2024
      */
-    public UserDTO getUserById(String idValue) {
-    	UserDTO user = null;
+    public User getUserById(String idValue) {
     	User data = null;
     	
     	try {
 			data = this.userRepository.findById(idValue).get();
```

Comment:
If the user dos not exists.It will throw a NullPointerException
you can return the Optional 
        Optional<User> user = userRepository.findById(userId);
and in the controller you can return the response based on if it is present or not
//Keep in mind that the map operation is not for the streams it is a map between Optional, it is converting from this
Optional<User> to this Optional<ResponseEntity<UserDTO>>
    return userService.getUserById(id)
            .map(user -> {
        log.info("User found: {}", user);
        return ResponseEntity.ok(modelMapper.map(user, UserResponse.class));
    })
            .orElseGet(() -> {
        log.warn("User not found with id: {}", id);
        return ResponseEntity.notFound().build();
    });